src/XComponent.Functions.Test/FunctionsManagerTest.cs

```diff
@@ -168,14 +169,54 @@ public interface ISender
                 PublicMember = publicMemberAfter,
             };
 
+            var sender = new object(); 
+            var context = new object();
             var functionsManager = new FunctionsManager("component", "statemachine");
+            functionsManager.AddTask(null, null, null, context, sender);
```

Comment:
why?

Reply:
I get why you added that now, if sender is null you probably get an exeception somewhere...
please check if we are garanteed to get a sender object or if can be null (if there are no senders to call)